### PR TITLE
make sure intersecting gem paths are cached

### PR DESCRIPTION
**Pull request by indirect**
When a gem has the same path prefix as the application that is being bundled,
the gem will not be added to vendor/cache.

For example: a gem with a path /home/test/src/bundled_gem and an application with a path
/home/test/src/bundled that references it.

### CODE REVIEW

**Comment by indirect** [comment]
@homu r+

**Comment by indirect** [comment]
@homu r+

File: lib/bundler/source/path.rb [hunk 1]
```diff
@@ -77,7 +77,7 @@ def install(spec, force = false)
       def cache(spec, custom_path = nil)
         app_cache_path = app_cache_path(custom_path)
         return unless Bundler.settings[:cache_all]
-        return if expand(@original_path).to_s.index(Bundler.root.to_s) == 0
+        return if expand(@original_path).to_s.index(Bundler.root.to_s + '/') == 0
```
Comment (segiddins) [inline lib/bundler/source/path.rb]: double quotes

File: spec/cache/path_spec.rb [hunk 2]
```diff
@@ -29,6 +29,24 @@
       should_be_installed "foo 1.0"
     end
 
+    it "copies when the path is outside the bundle and the paths intersect" do
+      libname = File.basename(Dir.pwd) + '_gem'
```
Comment (segiddins) [inline spec/cache/path_spec.rb]: double quotes

**Comment by indirect** [comment]
@homu r+

**Comment by indirect** [comment]
@homu r+

**Comment by segiddins** [comment]
@homu r+

**Comment by indirect** [comment]
@homu retry

**Pull request merged**